controllers/datafoundation.go

```diff
@@ -214,9 +236,42 @@ func (r *dataFoundationReconciler) setDeviceSetCount(deviceSet *ocsv1.StorageDev
 		deviceSet.Count = currDeviceSetCount
 	}
 }
+
+func (r *dataFoundationReconciler) reconcileCSV() error {
+	r.Log.Info("Reconciling OCS Operator CSV")
+
+	csv, err := r.getCSVByPrefix(ocsOperatorName)
+	if err != nil {
+		return fmt.Errorf("unable to find OCS CSV: %v", err)
+	}
+	err = r.GetAndUpdate(csv, func() error {
+		if err := r.own(csv, false); err != nil {
+			return fmt.Errorf("unable to set ownerRef on ocs csv: %v", err)
+		}
+		deployments := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+		for i := range deployments {
+			containers := deployments[i].Spec.Template.Spec.Containers
+			for j := range containers {
+				container := &containers[j]
+				resources := datafoundation.GetResourceRequirements(container.Name)
+				if !equality.Semantic.DeepEqual(container.Resources, resources) {
+					container.Resources = resources
+				} else {
+					r.Log.V(-1).Info("Could not find resource requirement", "Resource", container.Name)
+				}
```

Comment:
The `else` content doesn't match the `if` content.  Also, why do we need a check for  `DeepEqual` if `GetAndUpdate` already takes care of it?